kubeflow/trainer/rhai/transformers.py

```
@@ -463,6 +471,195 @@ def relative_update(self, inc=1):
+
+                    # Update progress
+                    file_size = os.path.getsize(local_file)
+                    with lock:
+                        uploaded_bytes += file_size
+                        if time.time() - last_update >= 1:
+                            mb = uploaded_bytes / (1024 * 1024)
+                            pct = int((uploaded_bytes / total_size) * 100) if total_size else 0
+                            _log(f"Progress: {mb:.1f}/{total_size / (1024 * 1024):.1f} MB ({pct}%)")
+                            last_update = time.time()
+
+                    return True
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)